lib/plausible_web/controllers/api/external_stats_controller.ex

```diff
@@ -11,12 +11,13 @@ defmodule PlausibleWeb.Api.ExternalStatsController do
   end
 
   def aggregate(conn, params) do
-    site = conn.assigns[:site]
+    site = conn.assigns[:site] |> Repo.preload(:owner)
```

Comment:
```suggestion
    site = Repo.preload(conn.assigns.site, :owner)
```

perhaps, because `site` assign is guaranteed by `AuthorizeStatsApiPlug`?

Reply:
Makes sense, thanks! https://github.com/plausible/analytics/pull/3670/commits/f6092be85b15e166e38a5a9019939bf31c9f2824